pkg/ebpf/event.go

```diff
@@ -86,6 +86,36 @@ func (cs *ConnStatsWithTimestamp) isExpired(latestTime uint64, timeout uint64) b
 	return latestTime > timeout+uint64(cs.timestamp)
 }
 
+func newBatch(data []byte) *batch {
+	return (*batch)(unsafe.Pointer(&data[0]))
+}
+
+// ExtractBatchInto extract network.ConnectionStats objects from the given `batch` into the supplied `buffer`.
+// The `start` (inclusive) and `end` (exclusive) arguments represent the offsets of the connections we're interested in.
+func ExtractBatchInto(buffer []network.ConnectionStats, b *batch, start, end int) []network.ConnectionStats {
+	if start >= end || end > TCPCloseBatchSize {
```

Comment:
Can we make sure `current  + connSize < C.sizeof_batch_t` so we're not relying on `TCPCCloseBatchSize` ?

Reply:
I've opted for changing how constant declaration is declared instead (since we used it in different places). Let me know if that's OK